README.md

```diff
@@ -324,6 +325,33 @@ target_link_libraries(my-executable PRIVATE adobe-contract-checks)
   implementations should call the `check_invariant()`(s) of their
   parent classes.
 
+  If your mutating function has multiple returns, it might be useful
+  to wrap your `check_invariant()` call in
+  [`boost::scope_success`](https://www.boost.org/doc/libs/master/libs/scope/doc/html/scope/scope_guards.html#scope.scope_guards.conditional)
+  (which also [may be
+  available](https://en.cppreference.com/w/cpp/experimental/scope_success)
+  in your implementation's `std::experimental` namespace), at the beginning
+  of the function.
+
+  ```c++
+  void mutate_me() {
+    std::experimental::scope_success _([&]{ check_invariant(); });
+    ...
+  }
+  ```
+
+  **Caveats:**
+
+  1. You still need to call `check_invariant()` anywhere the object
+     under mutation may exposed outside the class other than by
+     returning, e.g. if it is passed to a callback function.
+
+  2. If your function reports errors to the caller other than by
```

Comment:
That is not true - `scope_success` only checks on non-exception exit (hense "success").

Reply:
You missed the words "other than."  I'm talking about functions that e.g. return error codes.